src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java

```diff
@@ -28,15 +30,53 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
             playerMsg(TFM_Command.PLAYER_NOT_FOUND);
             return true;
         }
+        else if (args.length > 1)
+        {
+            final String reason = StringUtils.join(ArrayUtils.subarray(args, 1, args.length), " ");
+            smite(sender, player, reason);
+            return true;
+        }
 
-        smite(player);
+        else
+        {
+            sender.sendMessage(ChatColor.GREEN + "You must supply a reason in order to smite someone.");
+            return true;
+        }
+    }
+
+    public static void smite(final CommandSender sender, final Player player, final String reason)
+    {
+        TFM_Util.bcastMsg(String.format("%s has been a naughty, naughty boy/girl.\nThey have been smitten!\n" + ChatColor.RED + "Reason: %s (%s)", player.getName(), reason, sender.getName()), ChatColor.RED);
+        String full = String.format(ChatColor.RED + "%s has been smitten for %s", player.getName(), reason + " (" + sender.getName() + ")");
 
-        return true;
+        //Deop
```

Comment:
Duplicate code from here to line 74. Make sure only one instance of this logic exists.